lib/matplotlib/style/core.py

```diff
@@ -66,22 +67,92 @@ def use(style):
     else:
         styles = style
 
+    styles = insert_parent_styles(styles)
+
     for style in styles:
         if not cbook.is_string_like(style):
             mpl.rcParams.update(style)
             continue
 
-        if style in library:
-            mpl.rcParams.update(library[style])
+        mpl.rcParams.update(get_style_parameters(style))
+
+
+def insert_parent_styles(styles):
+    """Returns a list with all the styles and parents styles called"""
+    tmp_styles = []
+    for style in styles:
+        style = _get_parent_styles_from_str_or_dict(style)
+        tmp_styles.extend(style)
+    return tmp_styles[:]
+
+
+def _get_parent_styles_from_str_or_dict(styles):
+    """Returns the parent styles from a single style or a dict"""
+    styles = [styles]
+    has_parent = True
+    visited_styles = []
+    temp_styles = styles[:]
+
+    while has_parent:
+        for style in styles:
+            if style in visited_styles:
+                continue
+            visited_styles.append(style)
+
+            if _is_dict_like(style):
+                style_dic = style
+            else:
+                style_dic = get_style_parameters(style)
+
+            has_parent = 'style' in style_dic
```

Comment:
In here, what I do is:
- Get children of the style to examine and put them in the `temp_styles`
- Then I replace the styles to be scanned by the `temp_styles`, so that I have all the brothers on a level.
- Then each of those styles is tested to be if it already has been scanned, and skipped unless it has not, meaning that even though earlier nodes are not scanned right away, they will be right after the child in the last generation of the current branch is scanned.

I am not sure this is quite clear, but I tried every scenario I could and this function always gave me the complete list of children (1 node with 3 children, 3 children each worked)